components/runtime/src/scala/MbArray.scala

```diff
@@ -64,6 +64,14 @@ object MbArray {
   def empty[T](size: Int): MbArray[T] =
     new miniboxing.runtime.array.MbArray_L[T](size);
 
+  /** Element-by-element constructor */
+  def apply[T](xs: T*): MbArray[T] = {
+    val mb_array = MbArray.empty[T](xs.length)
+    var i = 0
+    for (x <- xs.iterator) { mb_array.update(i, x); i += 1 }
+    mb_array
+  }
+
```

Comment:
Milos, the problem with this version of `apply` is that it will always create the generic version of the `MbArray`, specifically `MbArray_L`. To create specialized arrays, we have two options:
 * either redirect `apply` to an optimized version, as we do for `empty` and `clone` ([this code](https://github.com/milosstojanovic/miniboxing-plugin/blob/issue214/components/plugin/src/miniboxing/plugin/transform/minibox/commit/MiniboxCommitTreeTransformer.scala#L256-L280) redirects to [the optimized implementations](https://github.com/milosstojanovic/miniboxing-plugin/blob/issue214/components/runtime/src/miniboxing/runtime/array/MbArrayOpts.java#L140-L206))
 * add the `@miniboxed` annotation to the `apply` method, which should specialize `MbArray.empty` to create the correct version (I personally don't like this option, as it adds to the recursive dependencies, which are already quite heavy)

Also, please add the following testcase:
```scala
println(MbArray(100,200,300).getClass) // should print "MbArray_I"
println(MbArray(1.0,2.0,3.0).getClass) // should print "MbArray_D"
println(MbArray("1", "2", "3").getClass) // should print "MbArray_L"
```